middlewares/accesslog/logger.go

```diff
@@ -139,6 +140,22 @@ func (l *LogHandler) Close() error {
 	return l.file.Close()
 }
 
+// Rotate closes and reopens the log file to allow for rotation
+// by an external source.
+func (l *LogHandler) Rotate() error {
+	var err error
+	if err = l.Close(); err != nil {
+		return fmt.Errorf("unable to close file: %s", err)
```

Comment:
What does the error returned by `l.Close()` look like? Often, stdlib's file-related error messages already contain the file name and the problem, so we might not need the "unable to close file" part (and thereby possibly avoid stuttering).

If we need it, please mention the filename in the error message.